### PR TITLE
Fixed incorrectly viewed side panels

### LINKED ISSUE #3575: [BUG] Dark Reader Inverting Vivaldi Web Panels
### **Describe the bug**

As of the most recent update, it appears that Dark Reader is inverting sites loaded via Vivaldi's web panels, even if they are not on the inversion list.

### **Steps to reproduce**

Open Vivaldi and add a web panel for Reddit, set Dark Reader to "Invert only listed" in the Site List. Open Reddit via the web panel.

### **Expected behavior**

Prior to the most recent update, Dark Reader had no affect on web panels at all.

### **Actual behavior**

Dark Reader appears to always invert; have tested on Reddit, Spotify, and Gmail.

### **Screenshots**

Annotation 2020-09-02 164938

### **System Information:**
 - OS: Windows 10
 - Browser: Vivaldi v3.2.1967.47
 - Dark Reader version: 4.9.17

### LINKED ISSUE #3577: [BUG] Dark Reader Inverting Opera Messenger (Web Panels)
### **Describe the bug**

As of the most recent update, it appears that Dark Reader is inverting sites loaded via Opera's Messenger (web panels), even if they are not on the inversion list.

### **Steps to reproduce**

Open Opera, right click on the left sidebar and add a Messenger for Whatsapp/Telegram, set Dark Reader to "Invert only listed" in the Site List. Open Whatsapp/Telegram via the Messenger.

### **Expected behavior**

Prior to the most recent update, Dark Reader had no affect on Messegner (web panels) at all.

### **Actual behavior**

Dark Reader appears to always invert; have tested on Whatsapp, Telegram, Twitter and Instagram.

### **Screenshots**
Whatsapp:
Annotation 2020-09-03 110620

### **System Information:**
 - OS: Windows 10
 - Browser: Opera 70.0.3728.144
 - Dark Reader version: 4.9.17

### **Additional context**

The bug it's almost the same as [this](https://github.com/darkreader/darkreader/issues/3575) on Vivaldi.

### PR DESCRIPTION
**Pull request by Gusted**
- Check if the incoming data is possible a webpanel and get the data according to the webpanel.
- Resolves #3575 
- Resolves #3577 

TODO: as the incoming data is:
```
{id: "cbladblaljcdhigblildkdclibaedpoe", url: "https://www.reddit.com/", origin: "https://www.reddit.com"}
```

and somehow the current fetch-response doesn't like the id 🤷🏽 A work around is needed

### CODE REVIEW

File: src/background/tab-manager.ts [hunk 1]
```diff
@@ -25,9 +25,22 @@ export default class TabManager {
...
             if (port.name === 'tab') {
-                const tabId = port.sender.tab.id;
-                const frameId = port.sender.frameId;
-                const url = port.sender.url;
+
+                let tabId;
+                let frameId;
+                let url;
+                let tabURL;
+                if (port.sender.tab == undefined) {
```
Comment (alexanderby) [inline src/background/tab-manager.ts:33]: `== null` is preferred

File: src/background/tab-manager.ts [hunk 2]
```diff
@@ -25,9 +25,22 @@ export default class TabManager {
...
+                    tabId = port.sender.tab;
```
Comment (alexanderby) [inline src/background/tab-manager.ts:34]: Did you mean `port.sender.id`?

File: src/background/tab-manager.ts [hunk 3]
```diff
@@ -60,7 +73,7 @@ export default class TabManager {
 
                 // Using custom response due to Chrome and Firefox incompatibility
                 // Sometimes fetch error behaves like synchronous and sends `undefined`
-                const sendResponse = (response) => chrome.tabs.sendMessage(sender.tab.id, {type: 'fetch-response', id, ...response});
+                const sendResponse = (response) => chrome.tabs.sendMessage(sender.tab == undefined ? sender.id as any : sender.tab.id, {type: 'fetch-response', id, ...response});
```
Comment (alexanderby) [inline src/background/tab-manager.ts:76]: `== null` is preferred. Why do you cast` sender.id` to `any`, what is the actual value, is it a number? Maybe it's a type definition bug.

**Review by alexanderby — CHANGES_REQUESTED** [review]

File: src/background/tab-manager.ts [hunk 4]
```diff
@@ -25,9 +25,22 @@ export default class TabManager {
...
-                const frameId = port.sender.frameId;
-                const url = port.sender.url;
+
+                let tabId;
+                let frameId;
+                let url;
+                let tabURL;
+                if (port.sender.tab == undefined) {
+                    tabId = port.sender.tab;
+                    frameId = 0;
```
[the remark this answers is not in the document]
Comment (Gusted) [reply src/background/tab-manager.ts:35]: `{id: "cbladblaljcdhigblildkdclibaedpoe", url: "https://www.reddit.com/", origin: "https://www.reddit.com"}` You see a proper way? It's some weird feature of Opera And Vivaldi that doesn't seem to care about a lot.

File: src/background/tab-manager.ts (hunk 3 above)
Comment (Gusted) [reply src/background/tab-manager.ts:76]: It's a string, but that doesn't get accepted as type and also it's gives a error in the console for some reason

**Pull request merged**